azure-pipelines-product-construction-service.yml

```diff
@@ -179,33 +95,154 @@ stages:
+
+          - task: AzureCLI@2
+            inputs:
+              azureSubscription: $(serviceConnectionName)
+              scriptType: pscore
+              scriptLocation: inlineScript
+              inlineScript: |
+                New-Item -ItemType Directory -Path $(diffFolder)
+                $before = az containerapp show --name $(containerappName) -g $(resourceGroupName) --output json
+                Set-Content -Path $(diffFolder)/before.json -Value $before
+            displayName: Snapshot configuration (before)
+
+          - download: current
+            displayName: Download ProductConstructionService.Deployment
+            artifact: ProductConstructionService.Deployment     
+
+          - task: NuGetToolInstaller@1
+            displayName: Use NuGet
+            inputs:
+              versionSpec: 5.3.x
+
+          - powershell: |
+              . .\eng\common\tools.ps1
+              InitializeDotNetCli -install:$true
+              .\.dotnet\dotnet workload install aspire
+            displayName: Install .NET and Aspire Workload
+
+          - powershell: .\eng\common\build.ps1 -restore
+            displayName: Install .NET
+
+          # We'll need to give this service connection permission to get an auth token for PCS
+          - task: AzureCLI@2
+            inputs:
+              azureSubscription: $(serviceConnectionName)
+              scriptType: pscore
+              scriptLocation: inlineScript
+              inlineScript: |
+                $(Pipeline.Workspace)/ProductConstructionService.Deployment/ProductConstructionService.Deployment.exe `
+                  --subscriptionId $(subscriptionId) `
+                  --resourceGroupName $(resourceGroupName) `
+                  --containerRegistryName $(containerRegistryName) `
+                  --containerAppName $(containerappName) `
+                  --workspaceName $(containerappWorkspaceName) `
+                  --containerJobNames $(containerjobNames) `
+                  --newImageTag $(newDockerImageTag) `
+                  --imageName $(containerName) `
+                  --azCliPath "$(azCliPath)" `
+                  --isCi true `
+                  --entraAppId $(MaestroAppId)
+            displayName: Deploy container app
+
+          - task: AzureCLI@2
+            inputs:
+              azureSubscription: $(serviceConnectionName)
+              scriptType: pscore
+              scriptLocation: inlineScript
+              inlineScript: |
+                $after = az containerapp show --name $(containerappName) -g $(resourceGroupName) --output json
+                Set-Content -Path $(diffFolder)/after.json -Value $after
+            displayName: Snapshot configuration (after)
+
+          # git diff will set the exit code to 1, since the files are different, we have to manually set it back to 0
+          - powershell: |
+              $diff = git diff before.json after.json
+              $LASTEXITCODE = 0
```

Comment:
I just tried it locally, it doesn't set it to 0 for some reason

Reply:
Oh, the var is used for last exit code of external commands.

We could maybe just ignore it completely with `ignoreLASTEXITCODE: true` instead of overwriting it - https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/powershell-v2?view=azure-pipelines#syntax